src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -248,29 +251,36 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       transition(WorkflowRunning) using data.copy(startMode = Option(startMode))
   }
 
-  def callCompletedWhileRunning(callKey: OutputKey, outputs: CallOutputs, returnCode: Int, message: TerminalCallMessage): State = {
-    val nextState = awaitCallComplete(callKey, outputs, returnCode) match {
-      case Success(_) =>
-        if (isWorkflowDone) transition(WorkflowSucceeded) else startRunnableCalls()
-      case Failure(e) =>
-        logger.error(e.getMessage, e)
+  private def handleCallCompleted(callKey: OutputKey, outputs: CallOutputs, returnCode: Int, message: TerminalCallMessage): State = {
+    // Don't close over sender().
+    val currentSender = sender()
+    val completionWork = for {
+      // TODO These should be wrapped in a transaction so this happens atomically.
```

Comment:
This is not a change in the current Cromwell behavior, I'm just calling it out a bit more stridently now.  :smile:  And I'll actually make a ticket for it now.